apriltag_ros/src/continuous_detector.cpp

```diff
@@ -105,12 +178,34 @@ void ContinuousDetector::imageCallback (
   // Check for an empty image and publish an empty detection message
   if (image_rect->data.empty())
   {
-    AprilTagDetectionArray tag_detection_array;
-    tag_detection_array.header = image_rect->header;
-    tag_detections_publisher_.publish(tag_detection_array);
+    publishEmptyDetection(image_rect->header);
+    return;
+  }
+
+  // Transform the target's pose from base link to this camera's optical frame
+  if (!toOptical(base_frame_id_, image_rect->header.frame_id, tag_pose_))
+  {
+    // ROS_INFO_STREAM_THROTTLE(1.0, "Valid pose not set");
+    publishEmptyDetection(image_rect->header);
     return;
   }
 
+  // Set the camera info
+  camera_model_.fromCameraInfo(camera_info);
+  // compute the pixel values for the FOV pixel buffer (ie. resizing the FOV)
+  fov_pixel_buffer_height_ = (camera_info->height/2) - ((camera_info->height * fov_size_scaler_) / 2);
+  fov_pixel_buffer_width_ = (camera_info->width/2) - ((camera_info->width * fov_size_scaler_) / 2);
```

Comment:
```suggestion
  fov_pixel_buffer_width_ = (camera_info-> reducedResolution().width/2) - ((camera_info-> reducedResolution().width * fov_size_scaler_) / 2);
```

I missed this earlier, but this is really important. Everywhere in here that you check the image size from the camera model, you need to use `reducedResolution()`. This will account for the binning and ROI information in the camera info, which matches what happens when you do the projection.

See the comments about binning_x/binning_y and roi
http://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/CameraInfo.html

Reply:
oh ok had no idea, will change this

Reply:
@nlunscher-cpr does this work? I have tried implementing it but camera_info is just a ros message, I don't think it has a reducedResolution() method (or at least i'd be confused if it did).

Reply:
```suggestion
  fov_pixel_buffer_width_ = (camera_model_->reducedResolution().width/2) - ((camera_model_->reducedResolution().width * fov_size_scaler_) / 2);
```

oops my bad, this is the right one, i think, i didnt test it...

Reply:
http://docs.ros.org/en/diamondback/api/image_geometry/html/c++/classimage__geometry_1_1PinholeCameraModel.html#aafe4584d4cf5278f7d3aeeecc639b618

Reply:
Yes makes sense, thx